src/main/java/teammates/ui/controller/InstructorSearchPageData.java

```diff
@@ -58,4 +139,274 @@ public String getStudentRecordsLink(String courseId, StudentAttributes student){
+            String frCommentGiver = feedbackResponseCommentSearchResultBundle
+                                            .commentGiverTable.get(frc.getId().toString());
+            String creationTime = TimeHelper.formatTime(frc.createdAt);         
+            String link = getInstructorCommentsLink() + "&" + Const.ParamsNames.COURSE_ID + "=" 
+                              + frc.courseId + "#" + frc.getId();         
+            ElementTag editButton = createEditButton(link, Const.Tooltips.COMMENT_EDIT_IN_COMMENTS_PAGE);
+            
+            rows.add(new FeedbackResponseCommentRow(frCommentGiver, frc.commentText.getValue(), 
+                                                        creationTime, editButton));
+        } 
+        return rows;
+    }
+    
+    private List<StudentRow> createStudentRows(String courseId) {
+        List<StudentRow> rows = new ArrayList<StudentRow>();      
+        List<StudentAttributes> studentsInCourse = filterStudentsByCourse(courseId);
+        
+        for (StudentAttributes student : studentsInCourse) {
+            String viewPhotoLink = addUserIdToUrl(student.getPublicProfilePictureUrl());        
+            String actions = getStudentActions(student);
+
+            rows.add(new StudentRow(viewPhotoLink, student.section, student.team, student.name, student.email, actions));
+        }
+        return rows;
+    }
+    
+    /**
+     * Filters students from studentSearchResultBundle by course ID
+     * @param courseId 
+     * @return students whose course ID is equal to the courseId given in the parameter
+     */
+    private List<StudentAttributes> filterStudentsByCourse(String courseId) {
+        List<StudentAttributes> students = new ArrayList<StudentAttributes>();
+        
+        for (StudentAttributes student : studentSearchResultBundle.studentList) {
+            if (courseId.equals(student.course)) {
+                students.add(student);
+            }
+        }
+        return students;
+    }
+
+    private ElementTag createEditButton(String href, String tooltip) {
+        return new ElementTag("href", href, "title", tooltip);
+    }
+    
+    /**
+     * Returns HTML string for student action links - view, edit, delete, all records, add comment
+     * when an instructor searches for a student
+     * @param student Details of the student
+     * @return HTML output for the links
+     */
+    public String getStudentActions(StudentAttributes student) {
+        StringBuilder result = new StringBuilder();
+        InstructorAttributes instructor = studentSearchResultBundle.instructors.get(student.course);
+        String disabledStr = "";
+        
+        // View       
+        if (!instructor.isAllowedForPrivilege(student.section, Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS)) {
+            disabledStr = "disabled=\"disabled\"";
+        } else {
+            disabledStr = "";
+        }
```

Comment:
For the first option (View) it doesn't do anything but in the subsequent blocks (Edit, Add comment etc. ) the else will reset the value of disabledStr because the same variable is used. For the first one, I kept it for consistency.